File: problems/remove-linked-list-elements/recursion.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode RemoveElements(ListNode head, int val) {
        if (head is null) return null;
        if (head.val == val) return RemoveElements(head.next, val);

       head.next = RemoveElements(head.next, val);
       
       return head;    
    }
}

File: problems/swap-nodes-in-pairs/recursion.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def swapPairs(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if head == None:
            return None
        if head.next == None:
            return head
        next = head.next.next
        newHead = head.next
        newHead.next = head
        head.next = self.swapPairs(next)
        return newHead
